odxtools/writepdxfile.py

```diff
@@ -128,9 +164,14 @@ def write_pdx_file(
         jinja_env.globals["make_xml_attrib"] = make_xml_attrib
         jinja_env.globals["make_bool_xml_attrib"] = make_bool_xml_attrib
 
-        vars: dict[str, Any] = {}
-        vars["odxtools_version"] = odxtools.__version__
-        vars["database"] = database
+        jinja_vars: dict[str, Any] = {}
+        jinja_vars["odxtools_version"] = odxtools.__version__
+        jinja_vars["database"] = database
+
+        jinja_env.globals["set_category_docfrag"] = lambda cname, ctype: set_category_docfrag(
+            jinja_vars, cname, ctype)
+        jinja_env.globals["set_layer_docfrag"] = lambda lname: set_layer_docfrag(jinja_vars, lname)
```

Comment:
`from functools import partial`
```suggestion
        jinja_env.globals["set_layer_docfrag"] = partial(set_layer_docfrag, jinja_vars)
```

Reply:
I think this should stay with the `lambda`s because `partial` is much less known (I didn't know it) and it is IMO also less explicit...